src/lib/table/table.md

```diff
@@ -35,60 +48,89 @@ Here's a simple column definition with the name `'userName'`. The header cell co
 </ng-container>
 ```
 
-#### 2. Define the table's rows
-
-After defining your columns, provide the header and data row templates that will be rendered out by 
-the table. Each row needs to be given a list of the columns that it should contain. The order of 
-the names will define the order of the cells rendered. It is not required to provide a list of all 
-the defined column names, but only the ones that you want to have rendered.
+#### 3. Define the table's rows
 
-```html
-<mat-header-row *matHeaderRowDef="['userName', 'age']"></mat-header-row>
-<mat-row *matRowDef="let myRowData; columns: ['userName', 'age']"></mat-row>
-```
+Finally, once you have defined your columns, you need to tell the table which columns will be
+rendered in the header and data rows. 
 
-#### 3. Provide data
-
-The column and row definitions now capture _how_ data will render - all that's left is to provide 
-the data itself. For simple scenarios with client-side operations, `MatTableDataSource` offers a 
-quick and easy starting point. Simply create an instance of `MatTableDataSource` and set the items 
-to be displayed to the `data` property. For more advanced scenarios, applications will implement 
-one or more custom `DataSource` to capture more specific behaviors.
+To start, create a variable in your component that contains the list of the columns you want to
+render. 
 
 ```ts
-this.myDataSource = new MatTableDataSource();
-this.myDataSource.data = dataToRender;
+columnsToDisplay = ['userName', 'age'];
 ```
 
 ```html
-<mat-table [dataSource]=”myDataSource”>
-  ...
-</mat-table>
+<mat-header-row *matHeaderRowDef="columnsToDisplay"></mat-header-row>
+<mat-row *matRowDef="let myRowData; columns: columnsToDisplay"></mat-row>
 ```
 
+Note that this list of columns provided to the rows can be in any order, not necessary the order in
+which you wrote the column definitions. Also, you do not necessarily have to include every column
+that was defined in your template.
+
+This means that by changing your column list provided to the rows, you can easily re-order and
+include/exclude columns dynamically.  
+
+### Alternative ways to provide data
+
```

Comment:
Consider adding a little blurb to refresh the reader of what the original option is

> ### Alternative ways to provide data
> The simplest way to provide data to your table is by passing a data array. More complex use-cases may benefit from a more flexible approach. _(or something like that)_
>
> #### Observable stream of data arrays
> ...
>
> #### DataSource
> ...